drivers/mx25v/mx25v.c

```diff
@@ -0,0 +1,541 @@
+/*
+ * Copyright (C) 2016  OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mx25v
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the mx25v flash memory
+ *
+ * @author      Aurelien Gonce <aurelien.gonce@altran.com>
+ *
+ * @}
+ */
+#include <stddef.h>
+
+#include <board.h>
+#include <xtimer.h>
+#include <cpu.h>
+#include <mutex.h>
+#include <periph/spi.h>
+#include <periph/gpio.h>
+#include <mx25v.h>
+
+/**
+ * @brief MX25V instruction code
+ * @{
+ */
+#define MX25V_RDID          (uint8_t)(0x9F) /**< Read Identification */
+#define MX25V_RES           (uint8_t)(0xAB) /**< Read Electronic ID */
+#define MX25V_REMS          (uint8_t)(0x90) /**< Read Electronic & Device ID */
+#define MX25V_WRSR          (uint8_t)(0x01) /**< Write Status Register */
+#define MX25V_RDSR          (uint8_t)(0x05) /**< Read Status Register */
+#define MX25V_READ          (uint8_t)(0x03) /**< 1 x I/O */
+#define MX25V_FASTREAD      (uint8_t)(0x0B) /**< Fast read data */
+#define MX25V_DREAD         (uint8_t)(0x3B) /**< 1In/2 Out fast read */
+#define MX25V_RDSFDP        (uint8_t)(0x5A) /**< Read SFDP */
+#define MX25V_WREN          (uint8_t)(0x06) /**< Write Enable */
+#define MX25V_WRDI          (uint8_t)(0x04) /**< Write Disable */
+#define MX25V_PP            (uint8_t)(0x02) /**< page program */
+#define MX25V_SE            (uint8_t)(0x20) /**< Sector Erase */
+#define MX25V_BE            (uint8_t)(0xD8) /**< Block Erase */
+#define MX25V_CE            (uint8_t)(0x60) /**< Chip Erase) hex code: 60 or C7 */
+#define MX25V_DP            (uint8_t)(0xB9) /**< Deep Power Down */
+#define MX25V_RDP           (uint8_t)(0xAB) /**< Release form Deep Power Down */
+/** @} */
+
+/** Timeout value for waiting operation done */
+#define MX25V_WRITE_STATUS_TIME (uint32_t)(5000)        /* write status time: TYP. 5  ms */
+#define MX25V_SECTOR_ERASE_TIME (uint32_t)(60000)       /* sector erase time: TYP. 60 ms */
+#define MX25V_BLOCK_ERASE_TIME  (uint32_t)(1000000)     /* block erase time:  TYP. 1  s */
+#define MX25V_CHIP_ERASE_TIME   (uint32_t)(15000000)    /* chip erase time:   TYP. 7  s */
```

Comment:
Use the preprocessor macros (constants) SEC_IN_USEC and MS_IN_USEC from timex.h for these kinds of things, it happens too easily that a zero too many or too few slips in when writing literal constants like this.
```c
#define MX25V_SECTOR_ERASE_TIME (uint32_t)(60 * MS_IN_USEC) 
#define MX25V_CHIP_ERASE_TIME   (uint32_t)(15 * SEC_IN_USEC)
```